core/o11y/src/pretty.rs

```diff
@@ -35,6 +36,31 @@ impl<'a> std::fmt::Display for Bytes<'a> {
     }
 }
 
+impl<'a> Bytes<'a> {
```

Comment:
You’ll have to change the definition of the type to `pub struct Bytes<T>(pub T);` and then do `impl std::str::FromStr for Bytes<Vec<u8>>`.  But this will cause some other issues at use sites where Bytes object is created from a reference to Vec.